File: Easy/diderog/meta_diderogdefaultview/meta_diderogdefaultview.cs
/*
    Easy
    Copyright (C) 2020 Università degli Studi di Catania (www.unict.it)
    This program is free software: you can redistribute it and/or modify
    it under the terms of the GNU General Public License as published by
    the Free Software Foundation, either version 3 of the License, or
    (at your option) any later version.
    This program is distributed in the hope that it will be useful,
    but WITHOUT ANY WARRANTY; without even the implied warranty of
    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
    GNU General Public License for more details.
    You should have received a copy of the GNU General Public License
    along with this program.  If not, see <http://www.gnu.org/licenses/>.
*/

﻿using System;
using System.Data;
using metadatalibrary;
using metaeasylibrary;
//$CustomUsing$


namespace meta_diderogdefaultview
{
    public class Meta_diderogdefaultview : Meta_easydata 
	{
        public Meta_diderogdefaultview(DataAccess Conn, MetaDataDispatcher Dispatcher) :
            base(Conn, Dispatcher, "diderogdefaultview") {
				Name = "Didattica Erogata";
			EditTypes.Add("default");
			ListingTypes.Add("default");
			//$EditTypes$
        }

		private string[] mykey = new string[] {"aa","idcorsostudio"};

		public override string[] primaryKey() {
			return mykey;
		}

		//$SetDefault$

		//$Get_New_Row$

		//$IsValid$

		public override void DescribeColumns(DataTable T, string ListingType) {
			base.DescribeColumns(T, ListingType);

			foreach (DataColumn C in T.Columns) {
				DescribeAColumn(T, C.ColumnName, "", -1);
			}
			int nPos = 1;
					
			switch (ListingType) {
				case "default": {
						DescribeAColumn(T, "corsostudio_title", "Corso di studio", nPos++);
						DescribeAColumn(T, "diderog_inesaurimento", "Inesaurimento", nPos++);
						DescribeAColumn(T, "aa", "Anno Accademico", nPos++);
						break;
					}
					//$DescribeAColumn$
			}
		}

		//$GetSortings$

		//$GetStaticFilter$
		
		//$CustomCode$
    }
}

File: Easy/Portale/client/app_segreterie/metadata/diderog/meta_diderogdefaultview.js
(function() {

    var MetaData = window.appMeta.MetaSegreterieData;

    function meta_diderogdefaultview() {
        MetaData.apply(this, ["diderogdefaultview"]);
        this.name = 'meta_diderogdefaultview';
    }

    meta_diderogdefaultview.prototype = _.extend(
        new MetaData(),
        {
            constructor: meta_diderogdefaultview,
			superClass: MetaData.prototype,

			describeColumns: function (table, listType) {
				var nPos=1;
				var self = this;
				_.forEach(table.columns, function (c) {
					self.describeAColumn(table, c.name, '', null, -1, null);
				});
				switch (listType) {
					default:
						return this.superClass.describeColumns(table, listType);
					case 'default':
						this.describeAColumn(table, 'corsostudio_title', 'Denominazione Corso di studio', null, 1100, 1024);
						this.describeAColumn(table, 'corsostudio_annoistituz', 'Anno accademico di istituzione Corso di studio', null, 1600, null);
						this.describeAColumn(table, 'aa', 'Anno Accademico', null, 2000, 9);
						this.describeAColumn(table, 'sede_title', 'Identificativo', null, 3200, 1024);
						this.describeAColumn(table, 'diderog_inesaurimento', 'Inesaurimento', null, 4000, null);
//$objCalcFieldConfig_default$
						break;
//$objCalcFieldConfig$
				}
				return appMeta.Deferred("describeColumns").resolve();
			},


			//$setCaptions$

			primaryKey: function () {
				return ["aa", "idsede", "idcorsostudio"];
			},


			//$isValidFunction$

			//$getStaticFilter$

			//$getSorting$

        });

    window.appMeta.addMeta('diderogdefaultview', new meta_diderogdefaultview('diderogdefaultview'));

	}());
